lib/ClangImporter/ClangModuleDependencyScanner.cpp

```diff
@@ -175,18 +181,89 @@ void ClangImporter::recordModuleDependencies(
     // Swift frontend option for input file path (Foo.modulemap).
     swiftArgs.push_back(clangModuleDep.ClangModuleMapFile);
 
+    // Handle VFSOverlay.
+    if (!ctx.SearchPathOpts.VFSOverlayFiles.empty()) {
+      for (auto &overlay : ctx.SearchPathOpts.VFSOverlayFiles) {
+        swiftArgs.push_back("-vfsoverlay");
+        swiftArgs.push_back(overlay);
+      }
+    } else {
+      // HACK: find the -ivfsoverlay option from clang scanner and pass to
+      // swift.
+      bool addOption = false;
+      for (auto &arg : clangModuleDep.BuildArguments) {
+        if (addOption) {
+          swiftArgs.push_back("-vfsoverlay");
+          swiftArgs.push_back(arg);
+          addOption = false;
+        } else if (arg == "-ivfsoverlay")
+          addOption = true;
+      }
+    }
+
     // Add args reported by the scanner.
-    llvm::for_each(clangModuleDep.BuildArguments, addClangArg);
+
+    // Round-trip clang args to canonicalize and clear the options that swift
```

Comment:
Nit: What do you think of factoring out the code that does this from the overall flow of this method?

Reply:
I think this should go the other direction. All operations on clang arguments should be done this way. Maybe I will do a cleanup for all the string arg manipulation in swift to use this round trip instead.